nexus_id_spec.md

```diff
@@ -0,0 +1,147 @@
+# Nexus Channel Core - Nexus ID Specification
+
+Version: 0.9.0
+
+## Purpose and Background
+
+The Nexus ID is a globally-unique identifier that is *required* to be stored
+by all devices that implement Nexus Channel Core. It solves many challenges
+facing manufacturers of interoperable devices that may need to be managed by
+external software platforms.
+
+## Structure
+
+The Nexus ID is a 6-byte structure that consists of two parts:
+
+```
+ <-2 bytes-> <----4 bytes---->
++-----------+-----------------+
+| Authority |    Device ID    |
+| ID        |                 |
++-----------+-----------------+
+```
+
+* 2-byte 'Authority ID' indicating what party assigned the device ID
+* 4-byte 'Device ID' unique for all IDs assigned from a given authority
+
+Manufacturers are assigned a unique Authority ID by the maintainer of Nexus
+(currently Angaza). Manufacturers can then serialize a practically infinite
+number of Device IDs according to their own practices and preferences. 
+
+As long as all manufacturers guarantee that they are using their own Authority 
```

Comment:
Note: We plan to use and promote a common "PAYG" authority ID (or set of IDs) for the industry, I think - worth checking with @chadnorvell . 

(While it is *possible* for us to assign unique authority IDs to each manufacturer, I assume we won't try to do this by default, since we want all Nexus "PAYG IDs" on the Angaza platform to have the same authority ID - us - for uniqueness across all PAYG devices).

Reply:
(Yes, that makes sense because we still have a PAYG ID uniqueness constraint for registering a device on the platform. Will change this part as long as Chad is in agreement -- and still keep the suggestion about using the Device ID part of the Nexus ID only for the external ID because that goes along with this operational model.)

Reply:
Per our conversation today, we're reserving authority ID 0x0000 to map 1:1 with PAYG IDs (existing and future). We strongly encourage product manufacturers to use these IDs.

If a manufacturer absolutely insists on using their own ID number space, we can issue other authority IDs. This may be necessary for some of the "import a ton of existing non-metered devices" projects in the future. But allowing this in the general case requires us to solve the problem of accepting and parsing prefixed unit numbers across our software stack.

(One option to avoid making serial numbers for other authority IDs excessively long is to borrow the PaygOps alphabetical prefix: the 2-byte authority ID could be represented by 1 to 3 alphabetical characters in base-26. Then the rest of the ID could remain in the 6, 7, 8 digit range.)